Convert chipflow.toml parsing to pydantic

robtaylor opened the pull request:
Will close #32

gatecat commented on pytest.xml:
Is this file supposed to be committed?

robtaylor commented:
Nope, good catch!

On Tue, 18 Mar 2025 at 15:05, myrtle ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> On pytest.xml
> <https://github.com/ChipFlow/chipflow-lib/pull/56#discussion_r2001279314>:
>
> Is this file supposed to be committed?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ChipFlow/chipflow-lib/pull/56#pullrequestreview-2694961675>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAB4O4GM5GCY774RHU77WJ32VAY2XAVCNFSM6AAAAABZEEACJ6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMOJUHE3DCNRXGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

robtaylor commented:
I fixed the lint error that was causing the CI to fail. The issue was an unused 'config' variable in __init__.py. I modified the code to simply validate the config_dict without capturing the result, which fixes the F841 lint error. All checks are now passing\!
